### PR TITLE
bamcmp

### PR DESCRIPTION
**Pull request by kmurat1**
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [X] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Review by SPPearce — APPROVED** [review]

File: modules/bamcmp/functions.nf [hunk 1]
```diff
@@ -0,0 +1,78 @@
+//
```
Comment (mahesh-panchal) [inline modules/bamcmp/functions.nf:1]: This file should be removed.

File: modules/bamcmp/main.nf [hunk 2]
```diff
@@ -0,0 +1,48 @@
+def VERSION = '2.2'
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:6]: This is still using the old format for modules. Please update to the latest.

File: modules/bamcmp/main.nf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+process BAMCMP {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}"
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:10]: This should not be part of the code.

File: modules/bamcmp/main.nf [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+    conda (params.enable_conda ? "bioconda::bamcmp=2.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bamcmp:2.2--h05f6578_0"
+    } else {
+        container "quay.io/biocontainers/bamcmp:2.2--h05f6578_0"
+    }
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:18]: This needs to be reformatted to the latest version.

File: modules/bamcmp/main.nf [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    output:
+    tuple val(meta), path("*primary.bam"), emit: bam
+    tuple val(meta), path("*contamination.bam"), emit: contamination_bam
+    path "versions.yml"           , emit: versions
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:32]:
```suggestion
```

File: modules/bamcmp/main.nf [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        container "quay.io/biocontainers/bamcmp:2.2--h05f6578_0"
+    }
+
+    input:
+    tuple val(meta), path(sample), path(contaminant)
+
+    output:
+    tuple val(meta), path("*primary.bam"), emit: bam
+    tuple val(meta), path("*contamination.bam"), emit: contamination_bam
+    path "versions.yml"           , emit: versions
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:26]:
```suggestion
    tuple val(meta), path("*primary.bam")      , emit: bam
    tuple val(meta), path("*contamination.bam"), emit: contamination_bam
    path "versions.yml"                        , emit: versions
```

File: modules/bamcmp/main.nf [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+
+    """
+    bamcmp \\
+        -s "as" \\
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:35]: should this be part of the `$args` or is it mandatory?

File: modules/bamcmp/main.nf [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        -1 $sample \\
+        -2 $contaminant \\
+        -A ${prefix}_primary.bam \\
+        -B ${prefix}_contamination.bam
+
+cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        bamcmp: $VERSION
+    END_VERSIONS
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:44]:
```suggestion
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bamcmp: $VERSION
    END_VERSIONS
```

File: modules/bamcmp/main.nf [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (mahesh-panchal) [inline modules/bamcmp/main.nf:45]:
```suggestion
```

File: modules/bamcmp/main.nf (hunk 7 above)
Comment (mahesh-panchal) [reply modules/bamcmp/main.nf:35]: Also where is `$args` used here?

**Review by mahesh-panchal — CHANGES_REQUESTED** [review]

File: modules/bamcmp/main.nf (hunk 7 above)
Comment (kmurat1) [reply modules/bamcmp/main.nf]: Hi Mahesh, thank you for all your feedback and help with this PR. I hope my container part of code is up to date now. We strongly suggest using the "as" mode but I change it to be optional.

**Comment by mahesh-panchal** [comment]
Could you make sure you've followed all the instructions for contributing a module please?
The automated tests are not triggered, and my own running of the test shows it's failing.
I think something is still missing.

**Comment by mahesh-panchal** [comment]
The file `tests/config/pytest_modules.yml` should have been modified which would trigger the tests.

**Review by mahesh-panchal — APPROVED** [review]

**Pull request merged**